callstats-java-sdk/src/main/java/io/callstats/sdk/internal/CallStatsAuthenticator.java

```diff
@@ -167,6 +167,14 @@ public void doAuthentication() {
 		}
 	}
 	
+	public void doAuthenticationAfterExpire(long timeout) {
+		scheduler.schedule(new Runnable() {
+			public void run() {
+				doAuthentication();
+			}
+		}, timeout, TimeUnit.SECONDS);
+	}
```

Comment:
This should check if there are scheduled reauthentications and cancel them. Access to that object should be behind a lock.

Currently there are some code paths which can cause multiple authentication to happen (we can fix that at some point, but it's not critical) and thus multiple timers will be started (this should not happen as these can grow unbounded).